projects/plugins/crm/src/js/components/automations-admin/types.tsx

```diff
@@ -0,0 +1,37 @@
+export type Trigger = {
+	slug: string;
+	title: string;
+	description?: string;
+	category: Category;
+};
+
+export type Step = {
+	attributes: string[];
+	nextStep?: Step;
+	slug: string;
+	title: string;
+	description: string;
+	type: Type;
+	category: Category;
+	allowedTriggers: Trigger[];
+};
+
+export type Action = Step;
+
+export type Condition = Step;
+
+export type Workflow = {
+	id: number;
+	name: string;
+	description: string;
+	category: Category;
+	triggers: Trigger[];
+	initial_step: Step;
```

Comment:
Our initial step is being defined as an array, would this be a problem or is it ok to always send `initial_step[0]`?

```
	/**
	 * The workflow initial step.
	 *
	 * @since $$next-version$$
	 * @var array
	 */
	public $initial_step;

	/**
	 * Get the initial step of this workflow.
	 *
	 * @since $$next-version$$
	 *
	 * @return array
	 */
	public function get_initial_step(): array {
		return $this->initial_step;
	}
```

Reply:
Good catch, I changed this to be an array.

Reply:
Mmm here really the initial step is a Step instance. In the Workflow is defined as an array (as raw data) because is the way we are going to store it in the database (as JSON), but in this context, it could be wrapped as a Step instance.